src/Raven.Server/Documents/Queries/QueryBuilderHelper.cs

```diff
@@ -528,9 +528,10 @@ internal static QueryFieldName ExtractIndexFieldName(Query query, BlittableJsonR
     {
         RuntimeHelpers.EnsureSufficientExecutionStack();
         FieldMetadata metadata;
-
-        if (fieldName.Equals(Constants.Documents.Indexing.Fields.DocumentIdMethodName, StringComparison.OrdinalIgnoreCase) 
-            || fieldName is Constants.Documents.Indexing.Fields.DocumentIdFieldName)
+        
+        //Sometimes index can contians Id property and its different than real document ID. We've to 
+        if ((fieldName.Equals(Constants.Documents.Indexing.Fields.DocumentIdMethodName, StringComparison.OrdinalIgnoreCase) && indexMapping.ContainsField(fieldName)) == false && 
```

Comment:
You changed the operator and now instead of `||` you have `&&` between conditions. Can we have a test for that case?

Reply:
Done